src/components/Property/Types/VectorProperty/VectorProperty.tsx

```diff
@@ -20,11 +23,24 @@ export interface VectorPropertyProps {
 }
 
 export function VectorProperty(props: VectorPropertyProps) {
-  const { value, viewOptions } = props;
+  const { value, viewOptions, name, description } = props;
+
+  const isColor = (value.length === 3 || value.length === 4) && viewOptions.Color;
+  const isMinMaxRange = value.length === 2 && viewOptions.MinMaxRange;
```

Comment:
Why do we need to check for this here? Can't we use only viewOption as the determining factor and throw an error in the component (ColorVector, MinMaxRange) if it is wrong? As it is now we allow the property to be set wrongly in openspace

Reply:
This is simply copy-pasted from how it was implemented in the old UI, where it was simply a safeguard against if someone sets the wrong viewoption in C++ (which in general, we shouldn't), and cummincate in some way that this is the use case when each view option is supported on one line, I guess 😅  

But I agree, now that there are separate components for each view option, it would be nicer to check it in each component and throw of the number of values is wrong.  We should get away from the old tendency of silently ignoring errors